### PR TITLE
refacors s3object resource

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1918. Based on https://github.com/giantswarm/aws-operator/pull/1107.

### CODE REVIEW

File: service/controller/v15/key/key.go [hunk 1]
```diff
@@ -108,6 +108,8 @@ func BucketName(customObject v1alpha1.AWSConfig, accountID string) string {
 	return fmt.Sprintf("%s-g8s-%s", accountID, ClusterID(customObject))
 }
 
+// TODO what should the path look like when we want to use the version bundle
+// version of the operator?
 func BucketObjectName(templateVersion string, prefix string) string {
 	return fmt.Sprintf("cloudconfig/%s/%s", templateVersion, prefix)
 }
```
Comment (xh3b4sd) [inline service/controller/v15/key/key.go]: When I did not miss something critical this should be the place where we want to change the S3 object name. It should contain the version bundle version. How should the path look like though? I would provide a PR based on this one with further changes.

**Comment by xh3b4sd** [comment]
FYI canceled the `update` test since all this here is about to fix it. Just doesn't work yet due to the update issues of the S3 objects.

File: service/controller/v15/key/key.go (hunk 1 above)
Comment (rossf7) [reply service/controller/v15/key/key.go]: Yes as far as I recall this is the right place. The current paths with k8scloudconfig look like 

```
/cloudconfig/v_3_2_5/master
/cloudconfig/v_3_2_5/worker
```
To me adding the version bundle version but keeping the k8scloudconfig version makes sense.

```
/cloudconfig/3.4.0/v_3_2_5/master
/cloudconfig/3.4.0/v_3_2_5/worker
```

The S3 URL for small cloudconfig will be unique but if we change the k8scc version in a WIP release the paths will still be correct.

Comment (rossf7) [reply service/controller/v15/key/key.go]: Thought on this bit more and I think `/cloudconfig/3.4.0/cloudconfig/v_3_2_5/worker` is better.

It's useful that we version the S3 URLs and in future we might need to store other types of assets. WDYT?

File: service/controller/v15/resource/s3object/create.go [hunk 2]
```diff
@@ -33,34 +33,41 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 				return microerror.Mask(err)
 			}
 
-			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating S3 object '%s': created", key))
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created S3 object '%s'", key))
 		} else {
-			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating S3 object '%s': already created", key))
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not create S3 object '%s'", key))
```
Comment (kopiczko) [inline service/controller/v15/resource/s3object/create.go]: I think reason log would be useful here. Sth `S3 object '%s' already exists` would work for me.

File: service/controller/v15/key/key.go (hunk 1 above)
Comment (rossf7) [reply service/controller/v15/key/key.go]: Damn its too hot here! 😆 

Yes making the root `/version/` makes way more sense. What Pawel said. 😄

Comment (xh3b4sd) [reply service/controller/v15/key/key.go]: Cool. Thanks gang. Will address this separately.

**Comment by xh3b4sd** [comment]
CI was green. Merging into root feature branch and going from there again.

File: service/controller/v15/resource/s3object/create.go (hunk 2 above)
Comment (xh3b4sd) [reply service/controller/v15/resource/s3object/create.go]: Done.

**Pull request merged**